util/types/json/functions.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import "fmt"
+
+// Type returns type of JSON as string.
+func (j JSON) Type() string {
+	switch j.typeCode {
+	case typeCodeObject:
+		return "OBJECT"
+	case typeCodeArray:
+		return "ARRAY"
+	case typeCodeLiteral:
+		switch byte(j.i64) {
+		case jsonLiteralNil:
+			return "NULL"
+		default:
+			return "BOOLEAN"
+		}
+	case typeCodeInt64:
+		return "INTEGER"
+	case typeCodeFloat64:
+		return "DOUBLE"
+	case typeCodeString:
+		return "STRING"
+	default:
+		msg := fmt.Sprintf(unknownTypeCodeErrorMsg, j.typeCode)
+		panic(msg)
+	}
+}
+
+// Extract receives several path expressions as arguments, matches them in j, and returns:
+//  ret: target JSON matched any path expressions. maybe autowrapped as an array.
+//  found: true if any path expressions matched.
+func (j JSON) Extract(pathExprList []PathExpression) (ret JSON, found bool) {
+	elemList := make([]JSON, 0, len(pathExprList))
+	for _, pathExpr := range pathExprList {
+		elemList = append(elemList, extract(j, pathExpr)...)
+	}
+	if len(elemList) == 0 {
+		found = false
+	} else if len(pathExprList) == 1 && len(elemList) == 1 {
+		found = true
+		ret = elemList[0]
+	} else {
+		found = true
+		ret.typeCode = typeCodeArray
+		ret.array = append(ret.array, elemList...)
+	}
+	return
+}
+
+// Unquote is for JSON_UNQUOTE.
+func (j JSON) Unquote() string {
+	switch j.typeCode {
+	case typeCodeString:
+		return j.str
```

Comment:
for example,
```sql
select json_unquote(`"hello, world"`); -- should return `hello, world`
-- but j.String() will return a json marshal string,
-- in this case will be `"hello, world"`
```
so I use `j.str` instead of `j.String`.